server/server.go

```diff
@@ -103,10 +105,60 @@ type ArgoCDServerOpts struct {
 	RepoClientset   reposerver.Clientset
 }
 
+//defaultSettings sets default secret settings
+func defaultSettings(settingsMgr *settings_util.SettingsManager, opts ArgoCDServerOpts) (*settings_util.ArgoCDSettings, error) {
+	defaultString := ""
+	cdSettings, err := settingsMgr.GetSettings()
+	errors.CheckError(err)
+
+	if bytes.Compare(cdSettings.ServerSignature, []byte(defaultString)) == 0 {
+		// set JWT signature
+		signature, err := util_session.MakeSignature(32)
+		errors.CheckError(err)
+		cdSettings.ServerSignature = signature
+	}
+
+	if cdSettings.LocalUsers[common.ArgoCDAdminUsername] == "" {
+		//placeholder for when we replace this with get pod
+		passwordRaw := "password"
```

Comment:
@JazminGonzalez-Rivero @jessesuen I'm not entirely sure I understand this part.  I see the bit about it being a placeholder, but not sure what the ultimate aim is.  Might it be more secure to create the map below (line 128) with a hash of `"*"` (i.e., an invalid, unreachable hash; my understanding is that an asterisk is somewhat common for this) so that no login is possible until/unless the password becomes set?

Reply:
1. I think this logic might need to change following my previous suggestion to start out with empty secret. I believe settingsMgr.GetSettings() will only work when there is a fully populated settings.

2. lets name this function initializeSettings().

3. I actually just realized that we can use hostname as the default password. In kubernetes, this translates to the pod name. This way we can start out with sane defaults. 

That said, If the pod restarts before they have a chance to change the password, the new pod name will not reflect the password.